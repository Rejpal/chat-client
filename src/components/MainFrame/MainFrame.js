import React, { Component } from 'react'
import { observer, inject } from 'mobx-react'

import UserForm from './UserForm/UserForm'
import ReplyBox from './ReplyBox/ReplyBox'
import Message from './Message/Message'

export class MainFrame extends Component {
  renderMessages () {
    const messages = this.props.appStore.getMessages()
    const messageComponents = []
    messages.forEach((message) => {
      messageComponents.push(<Message message={message} key={message.id} />)
    })

    return messageComponents
  }

  render () {
    if (this.props.appStore.user.isNameEmpty()) {
      return <UserForm />
    }

    return (
      <React.Fragment>
        <div className='Messages'>
          {this.renderMessages()}
        </div>
        <ReplyBox />
      </React.Fragment>
    )
  }
}

export default inject('appStore')(observer(MainFrame))
