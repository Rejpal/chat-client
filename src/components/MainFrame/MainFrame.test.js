/* eslint-env mocha, jest */
import React from 'react'
import { MainFrame } from './MainFrame'
import { shallow } from 'enzyme'

describe('<MainFrame>', () => {
  it('should render when userName is not empty and there are no messages', () => {
    const appStore = {
      user: { isNameEmpty: () => { return false } },
      getMessages: () => { return [] }
    }
    expect(shallow(<MainFrame appStore={appStore} />)).toMatchSnapshot()
  })

  it('should render when userName is not empty and there are some messages', () => {
    const appStore = {
      user: { isNameEmpty: () => { return false } },
      getMessages: () => { return [{ id: '1' }, { id: '2' }] }
    }
    expect(shallow(<MainFrame appStore={appStore} />)).toMatchSnapshot()
  })

  it('should render userForm when userName is empty', () => {
    const appStore = {
      user: { isNameEmpty: () => { return true } }
    }
    expect(shallow(<MainFrame appStore={appStore} />)).toMatchSnapshot()
  })
})
