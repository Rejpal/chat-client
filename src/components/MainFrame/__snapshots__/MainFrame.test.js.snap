// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainFrame> should render userForm when userName is empty 1`] = `<inject-UserForm-with-appStore />`;

exports[`<MainFrame> should render when userName is not empty and there are no messages 1`] = `
<Fragment>
  <div
    className="Messages"
  />
  <inject-ReplyBox-with-appStore />
</Fragment>
`;

exports[`<MainFrame> should render when userName is not empty and there are some messages 1`] = `
<Fragment>
  <div
    className="Messages"
  >
    <Message
      key="1"
      message={
        Object {
          "id": "1",
        }
      }
    />
    <Message
      key="2"
      message={
        Object {
          "id": "2",
        }
      }
    />
  </div>
  <inject-ReplyBox-with-appStore />
</Fragment>
`;
